lib/application/auth/auth_bloc.dart

```diff
@@ -146,8 +146,8 @@ class AuthBloc extends Bloc<AuthEvent, AuthState> {
   ) async {
     emit(const AuthState.awaitingUsernameUpdate());
 
-    final failureOrSuccess =
-        await _authRepository.updateUsername(username: event.username);
+    final failureOrSuccess = await _authRepository.updateUsername(
```

Comment:
As you're adding UsernameBloc, the AuthBloc no longer needs to have this event nor any state related to it.
Also the AuthRepository no longer needs to have the `updateUsername` method, it can be removed.

Reply:
removed